src/states/elder/mod.rs

```diff
@@ -2045,6 +1942,17 @@ impl Approved for Elder {
         self.pfx_is_successfully_polled
     }
 
+    fn handle_accumulated_message(&mut self, digest: Digest256) -> Result<(), RoutingError> {
+        let routing_message =
+            if let Some(routing_message) = self.signed_messages_accumulator.get(&digest) {
```

Comment:
This should probably be `.remove`.
* It would avoid cloning
* Currently, this is going to be an ever growing cache which is probably not what we want.